Fix query on ROS_DISTRO so bundle doesn't suppress 'ROS_DISTRO not defined' error.

zmichaels11 opened the pull request:
Fetching the `ROS_DISTRO` environment variable was silently failing here with the `Could not find package` error when `ROS_DISTRO` was not set in the environment. The try-except block is catching the `KeyError` from `os.environ['ROS_DISTRO']` and skipping the `else` block where it logs and raises an error when `ROS_DISTRO` is not defined.

Signed-off-by: Zachary Michaels <zacmicha@amazon.com>

emersonknapp reviewed (approved):
nice - any way to add a test for this?

emersonknapp commented on colcon_ros_bundle/task/ros_bundle.py:
```diff
@@ -75,18 +75,12 @@ async def bundle(self):  # noqa D:102
+                    ros_distro=ros_distro))
+        else:
+            logger.error('ROS_DISTRO is not defined make sure to '
+                         'source your ROS environment.')
```
Should this still be raising an exception?

zmichaels11 replied:
IMO: it should, but that changes behavior of `colcon-ros-bundle` since before this commit the tool would never have thrown that error.

thomas-moulard replied:
what about raising an exception? are there really cases where we bundle something without ros?

zmichaels11 commented:
I updated the tests so they check this behavior since it looks like that was their intent.

I modified:
* `test_exclude_ros_base` to check that nothing is added to the `apt` install list. `args.ros_distribution` and `args.exclude_ros_base` were never checked. (Verified with `grep -rn "ros_distribution"` on `colcon-ros-bundle` and `colcon-bundle`.)
* `test_include_ros_base` checks that `ros-kinetic-ros-base` is added to the `apt` install list. This test is mainly unchanged.

thomas-moulard commented on test/test_catkin_bundle.py:
```diff
@@ -22,14 +22,13 @@ async def test_bundle():
         'system': MagicMock(),
-        'other': MagicMock()
+        'other': MagicMock(),
+        'apt': MagicMock()
```
alphasort?

thomas-moulard commented on test/test_catkin_bundle.py:
```diff
@@ -38,9 +37,11 @@ async def test_bundle():
-        wrapper().get_rule.side_effect = _get_rule_side_effect
-        wrapper().resolve.side_effect = _resolve_side_effect
-        await task.bundle()
+        with patch('os.environ') as environ:
```
monkeypatch is what you want : https://docs.pytest.org/en/stable/reference.html#std-fixture-monkeypatch
